examples/reward_LM_gsm8k/inference.py

```diff
@@ -0,0 +1,119 @@
+
+    if base_lm == "openai":
+        base_model = OpenAIModel("gpt-4-1106-preview", additional_prompt="ANSWER")
+    elif base_lm == "google":
+        base_model = BardCompletionModel("gemini-pro", additional_prompt="ANSWER")
+    elif base_lm == "anthropic":
+        base_model = ClaudeModel("claude-3-opus-20240229", additional_prompt="ANSWER")
+    elif base_lm == "hf":
+        base_model = HFModel(model_dir, model_dir, quantized=quantized)
+    elif base_lm == "exllama":
+        base_model = ExLlamaModel(model_dir, max_batch_size = n_sc, max_seq_length=5000, max_new_tokens=1000, lora_dir=None)
+
+    reward_tokenizer = None
+    assert reward_dir is not None, "Reward model path cannot be none for CoT + Reward"
+    if reward_lm == "openai":
+        raise NotImplementedError("GPTCompletionModel does not support reward model output")
+    elif reward_lm == "google":
+        raise NotImplementedError("BardCompletionModel does not support reward model output")
+    elif reward_lm == "anthropic":
+        raise NotImplementedError("ClaudeModel does not support reward model output")
+    elif reward_lm == "hf":
+        reward_model = HFModelReward(model_pth = reward_dir, tokenizer_pth = reward_dir,quantized=quantized_reward, device_map = "cuda:0")
+    elif reward_lm == "exllama":
+        reward_model = ExLlamaModel(reward_dir, max_batch_size = n_sc, max_seq_length=5000, lora_dir=None)
+    
+    
+    
+    with open(prompt) as f:
+        prompt = json.load(f)
+
+    reasoner = CoT_Reward_Reasoner(base_model, temperature=temperature, n_sc=n_sc, bs=batch_size, main_prompt = prompt)
+    utils_reward = Reward_CoT_Utils(reward_model = reward_model)
+    evaluator = GSM8KEvaluator(
+                 output_extractor=utils_reward.cot_reward_sc_extractor,
+                 answer_extractor=lambda x: utils.retrieve_answer_from_dataset(x["answer"]),
+                 init_prompt=prompt, # will update dynamically
+                 disable_log=False,
+                 disable_tqdm=False,
+                 sample_prompt_type="cot")
+    from datetime import datetime
+    log_dir =  f'logs/gsm8k_'\
+                        f'cot/'\
+                        f'{datetime.now().strftime("%m%d%Y-%H%M%S")}'
+    if base_lm == 'hf':
+        model_name= model_dir.split('/')[-1]
+    else:
+        model_name = base_lm
+    log_dir = log_dir + f'_{model_name}'
+    accuracy = evaluator.evaluate(reasoner, shuffle_prompt=True, num_shot=4, resume=resume, log_dir=log_dir)
+    print(f'accuracy: {accuracy:.4f}')
+    return 0
+
+if __name__ == '__main__':
+    fire.Fire(main)
+"""
+CUDA_VISIBLE_DEVICES=2 python examples/cot_gsm8k/inference.py \
+--model_dir $Gemma_ckpts \ 
```

Comment:
If this is not the command for this script, we should remove that to avoid confusion

Reply:
Fixed, added the correct command that is associated with this file